lucene/core/src/java/org/apache/lucene/codecs/lucene50/Lucene50PostingsFormat.java

```diff
@@ -389,6 +390,33 @@
 
   private final int minTermBlockSize;
   private final int maxTermBlockSize;
+  private final FSTLoadMode fstLoadMode;
+
+  /**
+   * An enum that allows to control if term index FSTs are loaded into memory or read off-heap
+   */
+  public enum FSTLoadMode {
+    /**
+     * Always read FSTs from disk.
+     */
+    OFF_HEAP,
+    /**
+     * Never read FSTs from disk ie. all fields FSTs are loaded into memory
+     */
+    ON_HEAP,
+    /**
+     * Always read FSTs from disk.
+     * An exception is made for ID fields in an IndexWriter context which are always loaded into memory.
+     * This is useful to guarantee best update performance even if a non MMapDirectory is used which required for
+     * the {@link FSTLoadMode#AUTO} mode is used.
```

Comment:
Two `is used` in the last sentence here?

Maybe include a warning here that this does not look at `Directory` type and may put FST off-heap even for buffered directory implementations?